kof-operator/internal/server/handlers/collector_metrics_handler.go

```diff
@@ -187,39 +188,64 @@ func collectMetrics(ctx context.Context, kubeClient *k8s.KubeClient) (PodMetrics
 	}
 
 	metrics := make(PodMetricsMap, len(podList.Items))
-	multiErr := []error{}
+	errs := make([]error, 0, len(podList.Items))
+	errsCh := make(chan error)
 
-	for _, pod := range podList.Items {
-		metrics[pod.Name] = utils.Metrics{}
-		podMetrics := metrics[pod.Name]
-
-		metricsPort, err := getMetricsPort(&pod)
-		if err != nil {
-			multiErr = append(multiErr, fmt.Errorf("failed to get metrics port: %v", err))
-			continue
-		}
+	wg := sync.WaitGroup{}
+	syncMap := sync.Map{}
 
-		if err := collectHealthMetrics(podMetrics, &pod); err != nil {
-			multiErr = append(multiErr, fmt.Errorf("failed to collect health metrics: %v", err))
-		}
-
-		if err := collectResourceMetrics(ctx, kubeClient, podMetrics, &pod); err != nil {
-			multiErr = append(multiErr, fmt.Errorf("failed to collect resource metrics: %v, podName: %s", err, pod.Name))
-		}
+	for _, pod := range podList.Items {
+		wg.Add(1)
+		go func(pod corev1.Pod) {
+			defer wg.Done()
+
+			podMetrics := utils.Metrics{}
+			defer syncMap.Store(pod.Name, podMetrics)
```

Comment:
This method looks messy, not clear how the data is populated, and two mixed sync approaches in use - channels and syncMap, please refactor it to a more clean code.